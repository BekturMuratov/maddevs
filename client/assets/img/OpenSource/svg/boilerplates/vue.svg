<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>