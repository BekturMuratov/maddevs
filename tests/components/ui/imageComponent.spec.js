import { render } from '@testing-library/vue';

import ImageComponent from '@/components/ui/image-component';

describe('Webp image component', () => {
  const props = {
    fileName: 'Marat',
    fileNameRetina: 'Marat@2x',
    alt: 'marat',
    width: '1000',
    height: '500'
  };

  test('is Vue\'s instance', () => {
    const { container } = render(ImageComponent, {
      props
    });

    expect(container).toMatchSnapshot();
  });

  test('correctly props data', () => {
    const { getByTestId } = render(ImageComponent, {
      props
    });
    const result = getByTestId('test-multi-image');
    expect(result.width).toBe(Number(props.width));
    expect(result.height).toBe(Number(props.height));
    expect(result.alt).toBe(props.alt);
  });

  test('Has class if sectionIsRequired equal true', () => {
    const { container } = render(ImageComponent, {
      props
    });
    expect(container.getElementsByTagName('picture')).toBeTruthy();
  });
});
