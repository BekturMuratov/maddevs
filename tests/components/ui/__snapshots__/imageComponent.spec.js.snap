// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webp image component is Vue's instance 1`] = `
<div>
  <picture>
    <source
      class="multi-image"
      data-srcset="[object Object] ,[object Object] 2x"
      type="image/webp"
    />
     
    <img
      alt="marat"
      class="multi-image img_lazy"
      data-src="[object Object]"
      data-srcset="[object Object] 2x"
      data-testid="test-multi-image"
      height="500"
      width="1000"
    />
  </picture>
</div>
`;
